### DISCUSSION TITLE
.NET MAUI updates in .NET 9 Preview 5

### DISCUSSION THREAD

**Discussion by davidortinau** [dotnet/maui#22977] [Announcements]
The team is continuing work on core fundamentals of the .NET MAUI SDK to improve overall product quality. This includes expanding test coverage, end to end scenario testing, and bug fixing.

Here's a summary of what's new in .NET MAUI in this preview release:

- [Updated Blazor Hybrid Templates](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/dotnetmaui.md#blazor-hybrid-updated-templates)
- [Android API 35 Beta 2 Support](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/dotnetmaui.md#android-api-35-beta-2)
- [Android - LLVM Marshalled Methods](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/dotnetmaui.md#llvm-marshalled-methods)
- [Android - Trimming Enhancements](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/dotnetmaui.md#trimming-enhancements)

.NET MAUI updates in .NET 9 Preview 5:
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/dotnetmaui.md) 
* [What's new in .NET MAUI in .NET 9](https://learn.microsoft.com/dotnet/maui/whats-new/dotnet-9) documentation.
* [GitHub Release](https://aka.ms/maui9p5)

.NET 9 Preview 5:
* [Discussion](https://aka.ms/dotnet/9/preview5)
* [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview5/README.md)

**Comment by hamiddd1980** [comment]
Thank you for your efforts . but after so many releases of MAUI , I (as a developer from Visual Basic to C# ,WinForm, WPF,ASP.NET,Blazor,...) regret to say that **this is the first product from Microsoft that has a poor support for Right To Left layout in languages** like Arabic ,... and it seems there is no plan to add full support for this capability and requirement:
#14038  : **Graphical objects should be drawn from Right To Left when FlowDirection is RightToLeft**
Thanks again .

**Comment by rogerbriggen** [comment]
I have again a problem with the xaml compiler, this time with CommunityToolkit.MAUI popups. Please see https://github.com/dotnet/maui/issues/23042

**Reply by rogerbriggen** [reply]
@davidortinau : Please consider adding a up-to-date know issue for the releases. It took me quite some time to create a repo to just find out it gets closed since you find the bug already. This is very frustrating. Also, Preview 4 could not be used because of wrong dependencies to dotnet, Preview 5 cannot be used because these simple things to not work. That is why you do not get other feedback and the quality of MAUI is really not where it should be. Now, waiting to Preview 6 until I can do further tests, hopefully I can at least compile my project.
